pkg/utils/defaulter/clusterresourceplacement_test.go

```diff
@@ -0,0 +1,315 @@
+package defaulter
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+func TestSetDefaultsClusterResourcePlacement(t *testing.T) {
+	tests := map[string]struct {
+		obj     *fleetv1beta1.ClusterResourcePlacement
+		wantObj *fleetv1beta1.ClusterResourcePlacement
+	}{
+		"ClusterResourcePlacement with nil placement Policy": {
```

Comment:
we don't care about non-nil or non-empty fields and you can try to test nil/empty multiple fields in one test case unless the default value depends, eg, for the "clientSideApply", the ServerSideApplyConfig field is not needed.